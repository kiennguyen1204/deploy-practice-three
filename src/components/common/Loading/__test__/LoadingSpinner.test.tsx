import { render, screen } from '@testing-library/react';

// components
import LoadingSpinner from 'components/common/Loading/index';

describe('LoadingSpinner component', () => {
  it('renders the loading spinner', () => {
    render(<LoadingSpinner />);

    const spinnerContainer = screen.getByTestId('spinner-container');
    expect(spinnerContainer).toBeInTheDocument();

    const loadingSpinner = screen.getByTestId('loading-spinner');
    expect(loadingSpinner).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<LoadingSpinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
