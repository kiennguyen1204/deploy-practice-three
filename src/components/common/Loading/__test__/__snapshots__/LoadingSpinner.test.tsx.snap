// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="spinner-container"
    data-testid="spinner-container"
  >
    <div
      class="loading-spinner"
      data-testid="loading-spinner"
    />
  </section>
</DocumentFragment>
`;
